api/src/test/java/com/epam/pipeline/security/saml/SAMLUserDetailsServiceImplTest.java

```diff
@@ -241,13 +245,58 @@ public void shouldThrowAuthorizationExceptionForBlockedUser() {
         userDetailsService.loadUserBySAML(credential);
     }
 
+    @Test(expected = LockedException.class)
+    @Transactional(propagation = Propagation.REQUIRES_NEW, rollbackFor = Exception.class)
+    public void shouldThrowAuthorizationExceptionForUserWithBlockedApplicationProperty1() {
+        blockUserUsingCredentials(SAML_ATTRIBUTE_BLOCKED_USER_VALUE_1);
+        userDetailsService.loadUserBySAML(credential);
+    }
+
+    @Test(expected = LockedException.class)
+    @Transactional(propagation = Propagation.REQUIRES_NEW, rollbackFor = Exception.class)
+    public void shouldThrowAuthorizationExceptionForUserWithBlockedApplicationProperty2() {
+        blockUserUsingCredentials(SAML_ATTRIBUTE_BLOCKED_USER_VALUE_2);
+        userDetailsService.loadUserBySAML(credential);
+    }
+
+    @Test(expected = LockedException.class)
+    @Transactional(propagation = Propagation.REQUIRES_NEW, rollbackFor = Exception.class)
+    public void shouldThrowAuthorizationExceptionForUserWithBlockedApplicationProperty3() {
+        blockUserUsingCredentials(SAML_ATTRIBUTE_BLOCKED_USER_VALUE_3);
+        userDetailsService.loadUserBySAML(credential);
+    }
+
+    @Test
+    @Transactional(propagation = Propagation.REQUIRES_NEW, rollbackFor = Exception.class)
+    public void shouldAcceptUserWithValidNotBlockedApplicationProperty() {
+        blockUserUsingCredentials(SAML_ATTRIBUTE_NOT_BLOCKED_USER_VALUE);
+        final UserContext actualUserContext = userDetailsService.loadUserBySAML(credential);
+        Assert.assertEquals(expectedUserContext.getUsername(), actualUserContext.getUsername());
+    }
+
+    @Test
+    @Transactional(propagation = Propagation.REQUIRES_NEW, rollbackFor = Exception.class)
```

Comment:
You can add `@Transactional` to test class instead of annotation each test individally